src/shared/components/community/pending-follows.tsx

```diff
@@ -0,0 +1,315 @@
+import { RouteComponentProps } from "inferno-router/dist/Route";
+import { IRoutePropsWithFetch } from "@utils/routes";
+import { InfernoNode } from "inferno";
+import { PendingFollow } from "@components/common/pending_follow";
+
+type ViewState = "Unread" | "All";
+
+type PendingFollowsData = RouteDataResponse<{
+  listPendingFollowsResponse: ListCommunityPendingFollowsResponse;
+}>;
+
+interface PendingFollowsState {
+  appsRes: RequestState<ListCommunityPendingFollowsResponse>;
+  isIsomorphic: boolean;
+}
+
+interface PendingFollowsProps {
+  view_state: ViewState;
+  cursor?: DirectionalCursor;
+}
+
+function registrationStateFromQuery(view?: string): ViewState {
+  switch (view) {
+    case "Unread":
+    case "All":
+      return view;
+    default:
+      return "Unread";
+  }
+}
+
+export function getPendingFollowsQueryParams(
+  source?: string,
+): PendingFollowsProps {
+  return getQueryParams<PendingFollowsProps>(
+    {
+      view_state: registrationStateFromQuery,
+      cursor: (cursor?: string) => cursor,
+    },
+    source,
+  );
+}
+
+export type PendingFollowsFetchConfig = IRoutePropsWithFetch<
+  PendingFollowsData,
+  Record<string, never>,
+  PendingFollowsProps
+>;
+
+type PendingFollowsRouteProps = RouteComponentProps<Record<string, never>> &
+  PendingFollowsProps;
+
+@scrollMixin
+export class PendingFollows extends Component<
+  PendingFollowsRouteProps,
+  PendingFollowsState
+> {
+  private isoData = setIsoData<PendingFollowsData>(this.context);
+  state: PendingFollowsState = {
+    appsRes: EMPTY_REQUEST,
+    isIsomorphic: false,
+  };
+
+  loadingSettled() {
+    return resourcesSettled([this.state.appsRes]);
+  }
+
+  constructor(props: any, context: any) {
+    super(props, context);
+
+    this.handlePageChange = this.handlePageChange.bind(this);
+    this.handleApproveFollower = this.handleApproveFollower.bind(this);
+
+    // Only fetch the data if coming from another route
+    if (FirstLoadService.isFirstLoad) {
+      this.state = {
+        ...this.state,
+        appsRes: this.isoData.routeData.listPendingFollowsResponse,
+        isIsomorphic: true,
+      };
+    }
+  }
+
+  async componentWillMount() {
+    if (!this.state.isIsomorphic && isBrowser()) {
+      await this.refetch(this.props);
+    }
+  }
+
+  componentWillReceiveProps(
+    nextProps: PendingFollowsRouteProps & { children?: InfernoNode },
+  ): void {
+    if (
+      nextProps.view_state !== this.props.view_state ||
+      nextProps.cursor !== this.props.cursor
+    ) {
+      this.refetch(nextProps);
+    }
+  }
+
+  get documentTitle(): string {
+    return `${I18NextService.i18n.t(
+      "community_pending_follows",
+    )} - ${this.isoData.siteRes.site_view.site.name}`;
+  }
+
+  render() {
+    const state = this.state.appsRes.state;
+    const pending = state === "success" && this.state.appsRes.data.items;
+    return (
+      <div className="registration-applications container-lg">
```

Comment:
Do a grep for `registration` to try and find all these cases and replace them with pending-follows.

Reply:
Done